src/DependencyInjection/CompilerPass/VerifyRectorServiceExistsCompilerPass.php

```diff
@@ -24,9 +25,15 @@ public function process(ContainerBuilder $containerBuilder): void
                 continue;
             }
 
+            if (!class_exists($class)) {
+                throw new ShouldNotHappenException(
+                    sprintf('Rector rule "%s" not found, please verify that the class exists.', $class)
+                );
+            }
+
             if (! is_a($class, RectorInterface::class, true)) {
                 throw new ShouldNotHappenException(
-                    sprintf('Rector rule "%s" not found, please verify that the rule exists', $class)
+                    sprintf('Rector rule "%s" should extend "%s" or implement "%s".', $class, AbstractRector::class, RectorInterface::class)
```

Comment:
technically it only need to implement `RectorInterface` but since its good practice to subclass `AbstractRector` I figured it would be worth mentioning explicitly, even if technically it does not provide additional value.

Reply:
That's not true. 
Check other classes that require `RectorInterface`. It's not PHP only.

Reply:
This need to be resolved too

Reply:
Not sure what needs to change here

Reply:
@TomasVotruba any pointers how this should look like, so we can merge this PR?